src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -93,5 +94,57 @@ public virtual bool CanParseValue(object value)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Checks that the provided value is a valid default value.
+        /// This method should not throw an exception.
+        /// </summary>
+        public virtual bool IsValidDefault(object value)
+        {
+            try
+            {
+                return ToAST(value) != null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Converts a value to an AST representation. This is necessary for introspection queries
+        /// to return the default values of this scalar type when used on input fields or field and directive arguments. This method may throw an exception
+        /// or return <see langword="null"/> for a failed conversion.
+        /// </summary>
+        public virtual IValue ToAST(object value)
+        {
+            try
+            {
+                var serialized = Serialize(value);
+                return serialized switch
+                {
+                    bool b => new BooleanValue(b),
+                    byte b => new IntValue(b),
+                    sbyte sb => new IntValue(sb),
+                    short s => new IntValue(s),
+                    ushort us => new IntValue(us),
+                    int i => new IntValue(i),
+                    uint ui => new LongValue(ui),
+                    long l => new LongValue(l),
+                    ulong ul => new BigIntValue(ul),
+                    BigInteger bi => new BigIntValue(bi),
+                    decimal d => new DecimalValue(d),
+                    float f => new FloatValue(f),
+                    double d => new FloatValue(d),
+                    string s => new StringValue(s),
+                    null => null,
+                    _ => new StringValue(value.ToString())
```

Comment:
I have some doubts about the correctness of such a default case, but so far I can’t say for sure.

Reply:
Here we could eliminate this default case and throw an exception (similar to as is thrown for custom input object types' ToAST method).  Custom scalars can override this method if their `Serialize` method does not return a `string`.

What do you think?

Reply:
> similar to as is thrown for custom input object types' ToAST method

I wrote that it is better to return that code and not throw an exception. Input object is just a hierarchy of key-value pairs, it can be calculated based on its parts (scalars leaves).

> Here we could eliminate this default case and throw an exception 

But for scalars, I would really throw an exception. We know nothing about custom scalar logic. What if `Serialize` returns `FileStream` ? Probably the author conceived some kind of clever serialization of the stream. He definitely doesn't want to see something like `"System.IO.FileStream"`.

Reply:
Throws exception now, similar to `InputObjectGraphType<T>.ToAST` default implementation, where a custom scalar needs to implement this manually if they are not using a simple type.

Reply:
Good.